### PR TITLE
Implement VerticalTextAlignment property in SearchBarHandlers

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Implement `VerticalTextAlignment` property in SearchBarHandlers.

Related issue https://github.com/dotnet/maui/issues/398

### Platforms Affected ### 

- Core
- iOS
- Android

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [x] Targets a single property for a single control (or intertwined few properties)
- [x] Adds the property to the appropriate interface
- [x] Avoids any changes not essential to the handler property
- [x] Adds the mapping to the PropertyMapper in the handler
- [x] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [x] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [x] Tags ported renderer methods with [PortHandler]
- [x] Adds an example of the property to the sample project (MainPage)
- [x] Adds the property to the stub class
- [x] Implements basic property tests in DeviceTests

### CODE REVIEW

File: src/Core/src/Handlers/SearchBar/SearchBarHandler.iOS.cs [hunk 1]
```diff
@@ -8,6 +8,7 @@ namespace Microsoft.Maui.Handlers
 	public partial class SearchBarHandler : AbstractViewHandler<ISearchBar, UISearchBar>
 	{
 		UITextField? _editor;
+
 		public UITextField? QueryEditor => _editor;
```
Comment (rachelkang) [inline src/Core/src/Handlers/SearchBar/SearchBarHandler.iOS.cs:12]: Why is it called "QueryEditor" and "_editor"? Shouldn't it be "entry"?

Comment (rachelkang) [reply src/Core/src/Handlers/SearchBar/SearchBarHandler.iOS.cs:12]: or just "textField" as before?

**Review by rachelkang — CHANGES_REQUESTED** [review]
Add sample implementation of VerticalTextAlignment in MainPage!

Android tests all passing
Still need to test on iOS

**Review by mattleibow — CHANGES_REQUESTED** [review]
Something is up that I don't really understand. The alignment is weird in iOS and I can't check Android as the SearchBar is broken.

It is not this PR as when I commented everything out in the handler and also tried with the non-AppCompat version it was the same. iOS is just weird since it starts at the center and can't move to the top.

| Android | Android Typing | iOS |
| :--: | :--: | :--: |
|  |  |  |

**Review by rmarinho — APPROVED** [review]
Seems to work fine

End
Start
Center

Android 
image

File: src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs [hunk 2]
```diff
@@ -147,6 +147,9 @@ public static void DisposeMenuItems(this AToolbar toolbar, IEnumerable<ToolbarIt
 
 		internal static void UpdateMenuItemIcon(IMauiContext mauiContext, IMenuItem menuItem, ToolbarItem toolBarItem, Color tintColor)
 		{
+			if (toolBarItem.IconImageSource == null)
```
Comment (PureWeen) [inline src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs:150]: I think the extension method should be modified to accept a null ImageSource and then it just creates a result with a null value that gets passed into the callback

Comment (rmarinho) [reply src/Controls/src/Core/Platform/Android/Extensions/ToolbarExtensions.cs:150]: Can you make a PR with that? 
I will merge this now just to fix running android

**Review by rachelkang — COMMENTED** [review]
aw I just missed the merge :( seconding Matt's comment regarding iOS though - start alignment seems off

Screenshot 2021-09-23 at 10 06 33 AM

File: src/Core/src/Handlers/SearchBar/SearchBarHandler.iOS.cs (hunk 1 above)
Comment (rachelkang) [reply src/Core/src/Handlers/SearchBar/SearchBarHandler.iOS.cs]: still curious about this naming scheme too

**Comment by rmarinho** [comment]
buhh i missed that Start failll  , let's app a new issue.

**Pull request merged**